export async function sendVerificationEmail(userId) {
    try {
        const response = await fetch(`/api/admin/users/sendVerificationEmail/userId=${userId}`);
        if (!response.ok) {
            throw new Error('Network response was not ok');
        }
        const data = await response.json();
        return data;
    } catch (error) {
        console.error('There was a problem with the fetch operation:', error);
        throw error;
    }
}

export async function sendPasswordResetEmail(userId) {
    try {
        const response = await fetch(`/api/admin/users/sendPasswordResetEmail/userId=${userId}`);
        if (!response.ok) {
            throw new Error('Network response was not ok');
        }
        const data = await response.json();
        return data;
    } catch (error) {
        console.error('There was a problem with the fetch operation:', error);
        throw error;
    }
}

export async function confirmUserAccount(userId) {
    try {
        const response = await fetch(`/api/admin/users/confirmUserAccount/userId=${userId}`);
        if (!response.ok) {
            throw new Error('Network response was not ok');
        }
        const data = await response.json();
        return data;
    } catch (error) {
        console.error('There was a problem with the fetch operation:', error);
        throw error;
    }
}

export async function getUserInfo(userId) {
    try {
        const response = await fetch(`/api/admin/users/getUserInfo/userId=${userId}`);
        if (!response.ok) {
            throw new Error('Network response was not ok');
        }
        const data = await response.json();
        return data;
    } catch (error) {
        console.error('There was a problem with the fetch operation:', error);
        throw error;
    }
}

export async function updateUserAccount(userId, data) {
    try {
        const response = await fetch(`/api/admin/users/updateUserAccount/userId=${userId}`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify(data),
        });
        if (!response.ok) {
            throw new Error('Network response was not ok');
        }
        const responseData = await response.json();
        return responseData;
    } catch (error) {
        console.error('There was a problem with the fetch operation:', error);
        throw error;
    }
}

export async function deleteUser(userId) {
    try {
        const response = await fetch(`/api/admin/users/deleteUserAccount/userId=${userId}`, { method: 'DELETE' });
        if (!response.ok) {
            throw new Error('Network response was not ok');
        }
        const data = await response.json();
        return data;
    } catch (error) {
        console.error('There was a problem with the fetch operation:', error);
        throw error;
    }
}